dagger/rest/api.nim

```diff
@@ -0,0 +1,199 @@
+## Nim-Dagger
+## Copyright (c) 2021 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+{.push raises: [Defect].}
+
+import std/sequtils
+
+import pkg/questionable
+import pkg/questionable/results
+import pkg/chronicles
+import pkg/chronos
+import pkg/presto
+import pkg/libp2p
+import pkg/stew/byteutils
+
+import pkg/libp2p/routing_record
+
+import ../node
+
+proc validate(
+  pattern: string,
+  value: string): int
+  {.gcsafe, raises: [Defect].} =
+  0
+
+proc encodeString(cid: type Cid): Result[string, cstring] =
+  ok($cid)
+
+proc decodeString(T: type Cid, value: string): Result[Cid, cstring] =
+  let cid = Cid.init(value)
+  if cid.isOk:
+    ok(cid.get())
+  else:
+    case cid.error
+    of CidError.Incorrect: err("Incorrect Cid")
+    of CidError.Unsupported: err("Unsupported Cid")
+    of CidError.Overrun: err("Overrun Cid")
+    else: err("Error parsing Cid")
+
+proc encodeString(peerId: PeerID): Result[string, cstring] =
+  ok($peerId)
+
+proc decodeString(T: type PeerID, value: string): Result[PeerID, cstring] =
+  let peer = PeerID.init(value)
+  if peer.isOk:
+    ok(peer.get())
+  else:
+    err(peer.error())
```

Comment:
```suggestion
  ? PeerId.init(value)
```

Reply:
Unless it's to convert to cstring, then you could use [mapErr](https://github.com/arnetheduck/nim-result/blob/7c5f2423fe6042adc667ce7630a5b13349a13654/results.nim#L348-L352)